File: DSA/VenikaPanwar_590013755/Day_76_Question_3.java
public class Day_76_Question_3 {

    static class TreeNode {
        int val;
        TreeNode left, right;
        TreeNode(int val) { this.val = val; }
    }

    // Counts nodes at given level (root is level 1)
    public static int countAtLevel(TreeNode root, int level) {
        if (root == null) return 0;
        if (level == 1) return 1;
        return countAtLevel(root.left, level - 1) + countAtLevel(root.right, level - 1);
    }

    public static void main(String[] args) {
        // Example tree: 1 / \ 2 3 /
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);
        root.left.left = new TreeNode(4);

        int level = 2;
        System.out.println("Nodes at level " + level + ": " + countAtLevel(root, level)); // Output: 2
    }
}

File: DSA/RATNA_KHANNA_590013141/DAY_76_QUESTION_3.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* left;
    struct Node* right;
};

struct Node* createNode(int data) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = data;
    newNode->left = newNode->right = NULL;
    return newNode;
}

int countNodesAtLevel(struct Node* root, int level) {
    if (root == NULL) {
        return 0; // No node at this level
    }

    if (level == 0) {
        return 1; // Found a node at the target level
    }

    // Recursively count nodes at the given level in both subtrees
    return countNodesAtLevel(root->left, level - 1) + countNodesAtLevel(root->right, level - 1);
}

int main() {
    struct Node* root = createNode(1);
    root->left = createNode(2);
    root->right = createNode(3);
    root->left->left = createNode(4);

    int level;
    printf("Enter the level: ");
    scanf("%d", &level);

    int result = countNodesAtLevel(root, level);
    printf("Number of nodes at level %d: %d\n", level, result);

    return 0;
}

